### PR TITLE
test IntervalView to RAI

### PR DESCRIPTION
**Pull request by bnorthan**
I added tests for copying an interval view of a larger image to another image.

### CODE REVIEW

**Comment by bnorthan** [comment]
I am getting an error when trying to match the copy op.  Looks like ```CopyRAI.initialize``` is using in() to determine the type.

``` java
UnaryHybridCF<IntervalView<UnsignedByteType>, RandomAccessibleInterval<UnsignedByteType>> copy =
			(UnaryHybridCF) Hybrids.unaryCF(ops, Ops.Copy.RAI.class,
				RandomAccessibleInterval.class, IntervalView.class);

```

**Comment by dietzc** [comment]
this is similar to https://github.com/imagej/imagej-ops/issues/353. You can try to match it against a super-class of `IntervalView`.

**Comment by bnorthan** [comment]
@dietzc It looks like it matches with CopyRAI, but then fails in the initialize because there is no ```in()``` yet.  I noticed that CopyRAI checks to see if ```out()``` is null, but not ```in()```.  

```
final T inType = Util.getTypeFromInterval(in());
```

**Comment by LeonYang5114** [comment]
You are right. The only thing needed to fix this is to add null check for the `in()`. 

But this could apply to many other ops, since they might be match with no  `in()` or `out()` in some future use cases, and they want to match some helper ops. Then we need to do the null check every time we use `in()` or `out()`, or even better, make the matcher smarter so that the matcher can handler this automatically.

**Comment by bnorthan** [comment]
@LeonYang5114, I don't mind fixing small things like this as I encounter them... however it seems that the method to deal with ```null``` inputs is still being determined in #352.   So I don't want to change anything until there are examples of how the checks should work.

**Comment by LeonYang5114** [comment]
I would suggest you to fix that if that's what hindering your work. I just don't think it looks very elegant but that's what we have been doing in some places (e.g. [HistogramCreate](https://github.com/imagej/imagej-ops/blob/master/src/main/java/net/imagej/ops/image/histogram/HistogramCreate.java#L63), [MapNeighborhood](https://github.com/imagej/imagej-ops/blob/master/src/main/java/net/imagej/ops/map/neighborhood/MapNeighborhood.java#L73), [DefaultMedian](https://github.com/imagej/imagej-ops/blob/master/src/main/java/net/imagej/ops/stats/DefaultMedian.java#L62), [DefaultPercentile](https://github.com/imagej/imagej-ops/blob/master/src/main/java/net/imagej/ops/stats/DefaultPercentile.java#L66)), and I think we would need to write more this kind of null check later.

**Comment by bnorthan** [comment]
I've got it working, but the code is longer then I'd like 18de342e23a37b992d552086cc191ceabbadb205.

I first tried like this... but the weird thing was that it did not work for the case where in() was not null. 

``` java
@Override
	public void initialize() {
		final Class<?> outTypeClass =
			out() == null ? Type.class : Util.getTypeFromInterval(out()).getClass();
		
		final Class<?> inTypeClass = 
				in()==null ? NativeType.class :Util.getTypeFromInterval(in()).getClass();
		
		final UnaryComputerOp<?, ?> typeComputer =
			Computers.unary(ops(), Ops.Copy.Type.class, outTypeClass, inTypeClass);
		mapComputer = RAIs.computer(ops(), Ops.Map.class, in(), typeComputer);
		createFunc = RAIs.function(ops(), Ops.Create.Img.class, in(), inTypeClass);
	}
```

**Comment by dietzc** [comment]
`Util.getTypeFromInterval(in()).getClass()` why do you ask for the class here and not directly pass the type to the `CopyOp` (as you have it?). I would do `Util.getTypeFromInterval(in()).createVariable()` and pass this if possible. just go for `Object`.

**Comment by bnorthan** [comment]
@dietzc, like this??  Or did I misunderstand you?
```
final Class<?> outTypeClass =
			out() == null ? Type.class : Util.getTypeFromInterval(out()).getClass();
		
		final Object inType = 
        in()==null ? NativeType.class :Util.getTypeFromInterval(in());

			final UnaryComputerOp<?, ?> typeComputer =
			Computers.unary(ops(), Ops.Copy.Type.class, outTypeClass, inType);
			mapComputer = RAIs.computer(ops(), Ops.Map.class, in(), typeComputer);
			createFunc = RAIs.function(ops(), Ops.Create.Img.class, in(), inType);
```

**Comment by LeonYang5114** [comment]
It looks good to me.

File: src/main/java/net/imagej/ops/copy/CopyRAI.java [hunk 1]
```diff
@@ -70,13 +71,17 @@
 
 	@Override
 	public void initialize() {
-		final Class<?> outTypeClass =
-			out() == null ? Type.class : Util.getTypeFromInterval(out()).getClass();
-		final T inType = Util.getTypeFromInterval(in());
-		final UnaryComputerOp<T, ?> typeComputer =
-			Computers.unary(ops(), Ops.Copy.Type.class, outTypeClass, inType);
+		final Class<?> outTypeClass = out() == null ? Type.class : Util
```
Comment (dietzc) [inline src/main/java/net/imagej/ops/copy/CopyRAI.java]: here we could also type on `Object`?

**Comment by ctrueden** [comment]
I split out the `CopyRAITest` auto-formatting to its own commit, and merged it! Thanks!

**Pull request merged**